tests/functional/disaster-recovery/metro-dr/test_app_failover_and_relocate_when_entire_one_zone_down.py

```diff
@@ -0,0 +1,378 @@
+import logging
+
+import pytest
+import time
+from concurrent.futures import ThreadPoolExecutor
+
+from ocs_ci.framework.pytest_customization.marks import tier4a, turquoise_squad
+from ocs_ci.framework import config
+from ocs_ci.ocs.acm.acm import AcmAddClusters, validate_cluster_import
+from ocs_ci.ocs.dr.dr_workload import validate_data_integrity
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.ocs.node import get_node_objs, wait_for_nodes_status
+from ocs_ci.helpers.dr_helpers import (
+    enable_fence,
+    enable_unfence,
+    get_fence_state,
+    failover,
+    relocate,
+    restore_backup,
+    create_backup_schedule,
+    set_current_primary_cluster_context,
+    set_current_secondary_cluster_context,
+    get_current_primary_cluster_name,
+    get_current_secondary_cluster_name,
+    get_passive_acm_index,
+    wait_for_all_resources_creation,
+    wait_for_all_resources_deletion,
+    gracefully_reboot_ocp_nodes,
+    verify_drpolicy_cli,
+    verify_restore_is_completed,
+    verify_fence_state,
+)
+from ocs_ci.helpers.dr_helpers_ui import (
+    check_cluster_status_on_acm_console,
+    failover_relocate_ui,
+)
+from ocs_ci.ocs.resources.pod import wait_for_pods_to_be_running
+from ocs_ci.ocs.utils import get_active_acm_index
+from ocs_ci.utility.utils import TimeoutSampler, ceph_health_check
+from ocs_ci.utility import vsphere
+
+logger = logging.getLogger(__name__)
+
+
+@tier4a
+@turquoise_squad
+class TestApplicationFailoverAndRelocateWhenZoneDown:
+    """
+    Failover and Relocate with one entire Zone down (Co-situated Hub Recovery)
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request, dr_workload):
+        """
+        If fenced, unfence the cluster and reboot nodes
```

Comment:
There are failure cases in which teardown() has to make sure that the cluster is ready for executing the next test. For example, when relocation/failover fails. Need to explore such failure cases and take required proper actions. This can be taken up in next PRs.